### DISCUSSION TITLE
Using migrate fresh with cascades

### DISCUSSION THREAD

**Discussion by MortenDHansen** [maize-tech/laravel-markable#8] [Q&A]
The migrations specify `->constrained()->cascadeOnUpdate()->cascadeOnDelete()` on foreignId, but this prevents using Laravels database refresh since table users can't be dropped.

In context of the package, what is the functionality of cascade to the foreign?

**Comment by riccardodallavia** [comment]
Hi @MortenDHansen, we are currently using the constrained relation in order to be able to automatically delete all marks related to a user who is being deleted (or whose key is being updated).
Could you please provide more insights on the issue you are currently having with this approach?
